### PR TITLE
Filter build-info and modules properties

### PR DESCRIPTION
**Pull request by Or-Geva**
- [ ] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/build-info) passed. If this feature is not already covered by the tests, I added new tests.
-----
Prior to this PR, build-info properties were filtered only for environment variables and system properties.
The PR moves all filter logic into one place and performs exclude/include logic for all properties, including:

1. Environment variables (including properties with `buildInfo.env` prefix)
2. System properties
3. Properties from file
4. Build-info modules properties
5. Exclude not just the key but also the value of properties

### CODE REVIEW

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/packageManager/PackageManagerUtils.java
Comment (yahavi) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/packageManager/PackageManagerUtils.java:26]: I almost sure that reference tokens are 64 characters fixed

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/BuildInfoExtractorUtils.java
Comment (yahavi) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/BuildInfoExtractorUtils.java:110]: Removing the filtering code from this method sounds very risky to me. Let's do the "get" and the "filter" logic in one place to enforce filtering in all flows.

**Review by yahavi — CHANGES_REQUESTED** [review]
Please don't merge this code before my final approval. I want to go through it again.

File: build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/extractor/GradleBuildInfoExtractor.java
Comment (yahavi) [inline build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/extractor/GradleBuildInfoExtractor.java:184]: Why is that?

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/packageManager/PackageManagerUtils.java
Comment (Or-Geva) [reply build-info-extractor/src/main/java/org/jfrog/build/extractor/packageManager/PackageManagerUtils.java:143]: Changed to :
```
return variableValue.startsWith(secretPrefix) && variableValue.length() >= secretMinimalLength;
```

File: build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/extractor/GradleBuildInfoExtractor.java
Comment (Or-Geva) [reply build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/extractor/GradleBuildInfoExtractor.java:184]: Looks like this is related to:
```
for backward compatibility for Artifactory 2.2.3
```

File: build-info-extractor/src/test/java/org/jfrog/build/extractor/packageManager/PackageManagerUtilsTest.java
Comment (Or-Geva) [reply build-info-extractor/src/test/java/org/jfrog/build/extractor/packageManager/PackageManagerUtilsTest.java:21]: Sure, it is already tested [here](https://github.com/jfrog/build-info/pull/716/files/c382d3faa3423cf8044e7452b48621574fdd0e64#diff-c907117809abd7afd5787cd69de06463dde16bdc75877599532a497f57956e38R97)

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/BuildInfoExtractorUtils.java
Comment (Or-Geva) [reply build-info-extractor/src/main/java/org/jfrog/build/extractor/BuildInfoExtractorUtils.java:110]: I understand the concern, but this is not the place to do filtering. According to the name of this function, it returns all environment variables. By moving the filter logic to a new location, we can filter not only the environment variables, but also modules. Putting the filter logic here would prevent us from filtering things unrelated to environment variables, so we would have to split the logic across the code, and that will cause a mess in the code.

Comment (Or-Geva) [reply build-info-extractor/src/main/java/org/jfrog/build/extractor/BuildInfoExtractorUtils.java:110]: Due to security risks, we will return the filter logic to here despite the above comment .

**Review by yahavi — CHANGES_REQUESTED** [review]

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/packageManager/PackageManagerUtils.java
Comment (yahavi) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/packageManager/PackageManagerUtils.java:79]:
```suggestion
       filterBuildInfoProperties(clientConfiguration, buildInfo, clientConfiguration.getLog());
```

**Pull request merged**
